yb-voyager/cmd/exportSchemaOptimizationReport.go

```diff
@@ -0,0 +1,257 @@
+		Title:                    "Removed Redundant Indexes",
+		Description:              "The following indexes were identified as redundant and removed. These indexes were fully covered by stronger indexes—indexes that share the same leading key columns (in order) and potentially include additional columns, making the redundant ones unnecessary.",
+		TableToRemovedIndexesMap: make(map[string][]string),
+	}
+}
+
+// NewAppliedShardingRecommendationChange creates a new AppliedShardingRecommendationChange with default values
+func NewAppliedShardingRecommendationChange(objectType string) *AppliedShardingRecommendationChange {
+	var title, description string
+	switch objectType {
+	case TABLE:
+		title = "Applied Sharding Recommendations to Tables"
+		description = "Sharding recommendations from the assessment have been applied to the tables to optimize data distribution and performance. Tables will be created as colocated automatically according to the target database configuration."
+	case MVIEW:
+		title = "Applied Sharding Recommendations to Materialized Views"
+		description = "Sharding recommendations from the assessment have been applied to the mviews to optimize data distribution and performance. MViews will be created as colocated automatically according to the target database configuration."
+	default:
+		title = "Applied Sharding Recommendations"
+		description = "Sharding recommendations from the assessment have been applied to optimize data distribution and performance."
+	}
+
+	return &AppliedShardingRecommendationChange{
+		Title:             title,
+		Description:       description,
+		ShardedObjects:    make([]string, 0),
+		CollocatedObjects: make([]string, 0),
+	}
+}
+
+// NewSchemaOptimizationReport creates a new SchemaOptimizationReport with the given metadata
+func NewSchemaOptimizationReport(voyagerVersion, dbName, dbSchema, dbVersion string) *SchemaOptimizationReport {
+	return &SchemaOptimizationReport{
+		VoyagerVersion:        voyagerVersion,
+		SourceDatabaseName:    dbName,
+		SourceDatabaseSchema:  dbSchema,
+		SourceDatabaseVersion: dbVersion,
+	}
+}
+
+//go:embed templates/schema_optimization_report.template
+var optimizationChangesTemplate []byte
+
+// generatePerformanceOptimizationReport generates an HTML report detailing performance optimization changes applied to the exported schema.
+// It reports the removal of redundant indexes and the application of sharding recommendations to tables and materialized views (mviews).
+// The report includes references to the relevant SQL files and lists the modified objects.
+// Parameters:
+//   - redundantIndexes: list of redundant index names that were removed.
+//   - tables: list of table names to which sharding recommendations were applied.
+//   - mviews: list of materialized view names to which sharding recommendations were applied.
+func generatePerformanceOptimizationReport(redundantIndexes []string, shardedTables []string, shardedMviews []string) error {
```

Comment:
can you share the console output in the PR description?

Reply:
added it